File: mpp/LabMppCurseWPF/LabMppCurseWPF/Models/Client/ValidatorClient.cs
using CurseMPP.Models;

namespace LabMppCurseWPF.Models
{	
	/// <summary>
	/// Validator pentru Client
	/// </summary>
	public class ValidatorClient : IValidator<Client>
	{
		/// <summary>
		/// Valideaza un client, arunca ValidationException daca una din conditii nu e respectata
		/// </summary>
		/// <param name="entity">Clientul de validat</param>
		public void Validate(Client entity)
		{
			var errorMessage = "";

			if (entity.Nume.CompareTo("")==0)
			{
				errorMessage += "Numele unui client nu poate fi vid! ";
			}

			if (entity.Prenume.CompareTo("")==0)
			{
				errorMessage += "Prenumele unui client nu poate fi vid! ";
			}

			if (errorMessage.CompareTo("") != 0)
			{
				throw new ValidationException(errorMessage);
			}

		}
	}
}

File: mpp/LabMPPCurse/Models/src/main/java/Models/Client/ValidatorClient.java
package Models.Client;

import Models.ValidationException;
import Models.Validator;

public class ValidatorClient implements Validator<Client> {
    @Override
    public void validate(Client entity) throws ValidationException {

        String msg = "";

        if (entity.getNume().compareTo("")==0)
        {
            msg += "Nume vid";
        }

        if (entity.getPrenume().compareTo("")==0)
        {
            msg += "Prenume vid";
        }

        if (msg.compareTo("") != 0)
        {
            throw new ValidationException(msg);
        }
    }
}
